rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt

```diff
@@ -214,7 +214,7 @@ data class EvalStmt(
     override fun execute(interpreter: InterpreterCore) {
             // Should I assign it one by one?
             val result = if (target.type().isArray() &&
-                    target.type().asArray().element.canBeAssigned(expression.type())) {
+                    target.type().asArray().element.canBeAssigned(expression.type()) && expression.type() !is ArrayType) {
```

Comment:
I remember that the hypotesis was to fix `target.type().asArray().element.canBeAssigned` and` interpreter.assignEachElement` why did you decide to follow another approach?

Reply:
That function deals with completely different things and generated multiple different errors. I preferred to change course since I could easily handle the case in the `coerce.kt`. I handle the case in the same manner as the other types.

Reply:
How you can note, the theory (what we had in mind) can be completely different from the reality, and it is for this reason that when the people aks me evaluation, I would prefer answering: I have to try.